conda/common/logic.py

```diff
@@ -29,26 +29,195 @@
 """
 from __future__ import absolute_import, division, print_function, unicode_literals
 
+from array import array
 from itertools import chain, combinations
-from logging import getLogger
+from logging import DEBUG, getLogger
 import pycosat
 
 from .compat import iteritems
+from .io import time_recorder
 
 log = getLogger(__name__)
 
 
+class ClauseList(object):
+    def __init__(self):
+        self._clause_list = []
+        self.append = self._clause_list.append
+        self.extend = self._clause_list.extend
+        self.get_clause_count = self._clause_list.__len__
+
+    def save_state(self):
```

Comment:
This is more generally called `state` instead of `length` (`__len__`) since for `ClauseArray` it doesn't correspond to the number of clauses, but just to the integer array length.